apps/crn-server/test/data-providers/contentful/teams.data-provider.test.ts

```diff
@@ -251,97 +253,291 @@ describe('Teams data provider', () => {
+                      total: 1,
+                      items: [getContentfulGraphqlTeamMembers()],
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        };
 
-      test('Should return team tools by default', async () => {
-        const team1 = getContentfulGraphqlTeam();
-        team1.toolsCollection!.items! = [];
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+          teamsCollection: {
+            total: 1,
+            items: [team],
+          },
+        });
 
-        const team2 = getContentfulGraphqlTeam();
-        team2.toolsCollection!.items! = tools;
+        const result = await teamDataProvider.fetch({});
 
-        const team3 = getContentfulGraphqlTeam();
-        team3.toolsCollection!.items! = tools;
+        expect(result).toEqual({
+          items: [expect.objectContaining({ members: 1 })],
+          total: 1,
+        });
+      });
+
+      test('should filter non onboarded users', async () => {
+        const team = {
+          ...getContentfulGraphqlTeam(),
+          linkedFrom: {
+            teamMembershipCollection: {
+              total: 3,
+              items: [
+                {
+                  role: 'Collaborating PI',
+                  inactiveSinceDate: null,
+                  linkedFrom: {
+                    usersCollection: {
+                      total: 1,
+                      items: [getContentfulGraphqlTeamMembers()],
+                    },
+                  },
+                },
+                {
+                  role: 'Key Personnel',
+                  inactiveSinceDate: null,
+                  linkedFrom: {
+                    usersCollection: {
+                      total: 1,
+                      items: [
+                        {
+                          ...getContentfulGraphqlTeamMembers(),
+                          onboarded: false,
+                        },
+                      ],
+                    },
+                  },
+                },
+                {
+                  role: 'Project Manager',
+                  inactiveSinceDate: null,
+                  linkedFrom: {
+                    usersCollection: {
+                      total: 1,
+                      items: [getContentfulGraphqlTeamMembers()],
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        };
 
         contentfulGraphqlClientMock.request.mockResolvedValueOnce({
           teamsCollection: {
-            total: 3,
-            items: [team1, team2, team3],
+            total: 1,
+            items: [team],
           },
         });
 
-        const result = await teamDataProvider.fetch({
-          take: 8,
-          skip: 0,
+        const result = await teamDataProvider.fetch({});
+
+        expect(result).toEqual({
+          items: [expect.objectContaining({ members: 2 })],
```

Comment:
i wonder if maybe you should not just rely on the count here but actual items returned. What if someone comes in, changes the implementation so something else gets filtered out and the count still matches so its still all green but a different team is actually returned - thats not what you want right?

Reply:
But this is the `fetch` method test, so it does return only the number, not the items. The `fetchBy` method test still checks the items returned https://github.com/yldio/asap-hub/blob/ASAP-62-reduce-data-teams/apps/crn-server/test/data-providers/contentful/teams.data-provider.test.ts#L694:L696